sdcm/utils/docker_utils.py

```diff
@@ -368,6 +368,22 @@ def unregister_container(instance: INodeWithContainerManager, name: str) -> None
         del instance._containers[name]
         LOGGER.debug("%s: container `%s' unregistered", instance, name)
 
+    @classmethod
+    def destroy_unregistered_containers(
+            cls, instance: INodeWithContainerManager, docker_client: DockerClient = None) -> None:
+        """Destroy, if any, containers that were created for the instance during previous test run(s)"""
+        docker_client = docker_client or cls.default_docker_client
+        filters = {
+            "label": [
+                f"TestId={instance.tags['TestId']}",
+                f"Name={instance.name}",
+            ]
+        }
+        containers = docker_client.containers.list(all=True, filters=filters)
+        for container in containers:
+            if container not in instance._containers.values():
```

Comment:
Would be really useful to print a message saying what we remove and why.
In case something gets removed mistakenly we will be able to detect the problem much faster.

Reply:
Agree.
Added a message for such containers, example:
```
DEBUG    sdcm.utils.docker_utils:docker_utils.py:386 Container 'dummy' labeled with 'nodeA' node is not registered for this node. It is likely a leftover container from previous test run and was destroyed.
```